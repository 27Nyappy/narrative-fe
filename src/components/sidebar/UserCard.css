@media screen and (max-width: 640px) {
  .UserCard {
    display: flex;
    justify-content: center;
    flex-direction: column;
    width: 35%;

    img {
      max-width: 50px;
      border-radius: 50%;
      align-self: center;
    }

    button {
      align-self: center;
      background-color: #247E7F;
      color: white;
      border: none;
      border-radius: 5px;
      padding: 5px;
      font-size: .75em;
      text-transform: uppercase;
      transition: all ease-in .1s;
      width: 70px;
      height: 25px;
    }
  
    button:hover {
      background-color: #9E221A;
      transform: scale(1.05);
      cursor: pointer;
    }
  
    h3 {
      align-self: center;
      margin-top: 5px;
      margin-bottom: 5px;
    }
  }
}
@media screen and (min-width: 641px) and (max-width: 1007px) {
  .UserCard {
    height: 20vh;
    display: flex;
    justify-content: center;
    flex-direction: column;
    background-color: purple;
  
    img {
      margin: 8px 0px 5px 0px;
      max-width: 100px;
      border-radius: 50%;
      align-self: center;
    }
  
    button {
      align-self: center;
      margin: 5px 0px 10px 0px;
      width: 50%;
      background-color: #247E7F;
      color: white;
      border: none;
      border-radius: 5px;
      padding: 5px;
      font-size: .75em;
      text-transform: uppercase;
      transition: all ease-in .1s;
    }
  
    button:hover {
      background-color: #9E221A;
      transform: scale(1.05);
      cursor: pointer;
    }
  
    h3 {
      align-self: center;
      margin-top: 5px;
    }
  }
}
@media screen and (min-width: 1008px) {
  .UserCard {
    height: 20vh;
    display: flex;
    justify-content: center;
    flex-direction: column;
  
    img {
      margin: 8px 0px 5px 0px;
      max-width: 100px;
      border-radius: 50%;
      align-self: center;
    }
  
    button {
      align-self: center;
      margin: 5px 0px 10px 0px;
      width: 50%;
      background-color: #247E7F;
      color: white;
      border: none;
      border-radius: 5px;
      padding: 5px;
      font-size: .75em;
      text-transform: uppercase;
      transition: all ease-in .1s;
    }
  
    button:hover {
      background-color: #9E221A;
      transform: scale(1.05);
      cursor: pointer;
    }
  
    h3 {
      align-self: center;
      margin-top: 5px;
    }
  }
}

