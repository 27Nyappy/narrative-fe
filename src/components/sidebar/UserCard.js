import React from 'react';
import PropTypes from 'prop-types';
import { useSignout } from '../../Auth0Provider';
import styles from './UserCard.css';

const UserCard = ({ userName, userImage }) => {
  const logout = useSignout();
  const handleLogOut = () => {
    logout({
      returnTo: 'http://localhost:7890/stories'
    });
  };

  return (
    <section className={styles.UserCard}>
      <img src={userImage} alt={userName} />
      <h3>{userName}</h3>
      <button onClick={handleLogOut}>Logout</button>
    </section>
  );
};

UserCard.propTypes = {
  userName: PropTypes.string.isRequired,
  userImage: PropTypes.string.isRequired
};

export default UserCard;