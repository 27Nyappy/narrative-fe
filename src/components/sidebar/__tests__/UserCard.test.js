import React from 'react';
import { shallow } from 'enzyme';
import UserCard from '../UserCard';

describe('AboutDeveloperCard component', () => {
  it('renders an about developer card', () => {
    const wrapper = shallow(<UserCard name="user" picture="user.jpg" handleLogout={() => true}/>);
    expect(wrapper).toMatchSnapshot();
  });
});