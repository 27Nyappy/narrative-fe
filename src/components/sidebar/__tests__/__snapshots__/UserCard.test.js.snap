// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutDeveloperCard component renders an about developer card 1`] = `
<section>
  <img
    alt="user"
    src="user.jpg"
  />
  <h3>
    user
  </h3>
  <button
    onClick={[Function]}
  />
</section>
`;
